### PR TITLE
Autocomplete cell editor

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div
+[style.width.px]="columnWidth"
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.html]: Looks like there's some indentation missing on these lines.

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        SkyAppTestModule
+      ]
+    });
+
+    fixture = TestBed.createComponent(SkyAgGridCellEditorAutocompleteComponent);
+    nativeElement = fixture.nativeElement;
+    component = fixture.componentInstance;
+  });
+
+  it('renders an autocomplete input', () => {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.spec.ts]: The test description should be "should render an autocomplete input" for consistency.

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        context: undefined,
+        $scope: undefined,
+        stopEditing: undefined,
+        eGridCell: undefined,
+        parseValue: undefined,
+        formatValue: undefined
+      };
+    });
+
+    it('initializes the SkyuxNumericCellEditorComponent properties', () => {
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.spec.ts]: Same as previous comment about using "should." Apply this to the rest of the tests.

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  private params: ICellEditorParams;
+
+  @ViewChild('skyCellEditorAutocomplete', {read: ElementRef})
+  public input: ElementRef;
+
+  constructor(private libResources: SkyLibResourcesService) { }
+
+  public agInit(params: ICellEditorParams) {
+    this.params = params;
+    this.currentSelection = this.params.value;
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.ts]: Should this code be more defensive, i.e. is it possible that something like `this.params.column` might be undefined, causing an error when calling `getActualWidth()`? I know this function is called by ag-Grid directly, so if they guarantee these properties will be set, then there's no change needed here.

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+[style.height.px]="rowHeight"
+class="sky-ag-grid-cell-editor-autocomplete">
```
Comment (Blackbaud-PaulCrowder) [inline src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.html]: Drop the `>` to a new line.

File: src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.ts (hunk 4 above)
Comment (Blackbaud-ErikaMcVey) [reply src/app/public/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.component.ts]: my understanding is that they should always be set, but I'm not seeing that explicitly stated anywhere so I will go ahead and make it more defensive to be safe

**Pull request merged**